aten/src/ATen/cuda/CublasHandlePool.cpp

```diff
@@ -34,7 +42,11 @@ cublasHandle_t getCurrentCUDABlasHandle() {
   // See: https://github.com/pytorch/pytorch/pull/22405
   // This thread local unique_ptrs will be destroyed when the thread terminates,
   // releasing its reserved handles back to the pool.
-  static auto pool = std::make_shared<CuBlasPoolType>();
+
+  // Use a leaky singleton for the pool following standard practice around
+  // singletons: https://isocpp.org/wiki/faq/ctors#construct-on-first-use-v2
+  static auto pool = std::shared_ptr<CuBlasPoolType>(new CuBlasPoolType(),
+      LeakyDeleter());
```

Comment:
If it must be a shared_ptr, than at least let's use lambda instead of C++03 functor.
```suggestion
  static std::shared_ptr<CuBlasPoolType> pool(new CuBlasPoolType(), [](CuBlasPoolType *p) {});
```